state/runtime/access_list.go

```diff
@@ -0,0 +1,77 @@
+package runtime
+
+import (
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type AccessList map[types.Address]map[types.Hash]struct{}
+
+func NewAccessList() *AccessList {
+	al := make(AccessList)
+
+	return &al
+}
+
+// Checks if  address is present within the access list.
+func (al *AccessList) ContainsAddress(address types.Address) bool {
+	_, ok := (*al)[address]
+
+	return ok
+}
+
+// Contains checks if a slot is present in an account.
+// Returns two boolean flags: `accountPresent` and `slotPresent`.
+func (al *AccessList) Contains(address types.Address, slot types.Hash) (bool, bool) {
+	_, addrPresent := (*al)[address]
+	_, slotPresent := (*al)[address][slot]
+
+	return addrPresent, slotPresent
+}
+
+// Copy creates an deep copy of provided AccessList.
+func (al *AccessList) Copy() *AccessList {
+	cp := make(AccessList, len(*al))
+
+	for addr, slotMap := range *al {
+		cpSlotMap := make(map[types.Hash]struct{}, len(slotMap))
+		for slotHash := range slotMap {
+			cpSlotMap[slotHash] = struct{}{}
+		}
+
+		cp[addr] = cpSlotMap
+	}
+
+	return &cp
+}
+
+// AddAddress adds an address to the access list
+// returns 'true' if the operation results in a change (i.e., the address was not already present in the list).
+func (al *AccessList) AddAddress(address types.Address) bool {
+	if _, exists := (*al)[address]; exists {
+		return false
+	}
+
+	(*al)[address] = make(map[types.Hash]struct{})
+
+	return true
+}
```

Comment:
WDYT (it would be easier to invoke it, no need for the for loops when `ActivePrecompiles` are being added for instance etc.).

Also I'm not sure do we need to return a boolean, since its return value is not used anywhere.
```suggestion
func (al *AccessList) AddAddresses(addresses ...types.Address) {
	for _, addr := range address {
		if _, exists := (*al)[addr]; exists {
			continue
		}

		(*al)[addr] = make(map[types.Hash]struct{})
	}
}
```